specification/billing/resource-manager/Microsoft.Billing/preview/2018-05-31-preview/examples/BillingAccountsList.json

```diff
@@ -0,0 +1,38 @@
+{
+  "parameters": {
+    "api-version": "2018-05-31-preview",
+    "$filter": "permission eq 'createProject'"
```

Comment:
It seems odd and unnecessary to 1) support property expressions for properties *not* part of the resource being queried and 2) use a binary expression for something that is really a list. I would've expected something like "$filter": "hasPermission('createProject')", where hasPermission is a custom function.

@ravbhatnagar Is there any precedent for using properties in filter expressions that are not part of the schema? (In this case "permission" is not a property on the BillingAccount resource.)

Reply:
@wilcobmsft Do you have any example/existing swagger with similar scenario?

Reply:
Yes, look at Microsoft.Authorization. E.g. {scope}/Microsoft.Authorization/roleAssignments?assignedTo("my principal id") will return all role assignments that apply to you, even if you only have an indirect role assignment via a security group.